### ISSUE TITLE
aws_cloudwatch_metric_alarm should expsose arn

### ISSUE THREAD

**Issue by farrellit** [terraform-providers/terraform-provider-aws#5470]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Like nearly all aws resources, aws cloudwatch alarms have an Arn, and that Arn is occasionally useful, notably : 
* granting the alarm access to SNS / SQS with a condition `AWS::SourceArn`
* granting things access to the alarm, eg an IAM policy with the resource Arn.  This isn't supported today, but could very well be in the future.  
### New or Affected Resource(s)

* aws_cloudwatch_metric_alarm

### Potential Terraform Configuration
         

Currently I have to use 

```
"arn:aws:cloudwatch:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:alarm:${aws_cloudwatch_metric_alarm.alarm_name.id}"`
```

Ideally, I'd like it to be simply:

`"${aws_cloudwatch_metric_alarm.alarm_name.arn}"`

Complete resource example: 

```hcl
data "aws_iam_policy_document" "alarm-alert-policy" {
  policy_id = "__default_policy_ID"
  statement {
    actions = [ "SNS:Publish" ]
    condition = [
      {
        test     = "StringEquals"
        variable = "AWS:SourceArn"
        values = [  "arn:aws:cloudwatch:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:alarm:${aws_cloudwatch_metric_alarm.alarm_name.id}"
        ]
      }
    ]
    effect = "Allow"
    principals {
      type        = "AWS"
      identifiers = ["*"]
    }
    resources = [
      "${aws_sns_topic.topic_name.arn}",
    ]
    sid = "__default_statement_ID"
  }
}
```

### References

* #0000

**Comment by bflad** [comment]
Support for an `arn` attribute has been merged into master and will release with version 1.32.0 of the AWS provider, likely middle of next week.

**Comment by farrellit** [comment]
fantastic.  Impressive turnaround @bflad

**Comment by bflad** [comment]
This has been released in [version 1.32.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1320-august-16-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
